File: Python/Problem_076.py
def euler_76(n: int = 100) -> int:
    dp = [0] * (n + 1)
    dp[0] = 1

    for part in range(1, n):          # 1..99
        for s in range(part, n + 1):  # part..100
            dp[s] += dp[s - part]

    return dp[n]

if __name__ == "__main__":
    print(euler_76())  # 190569291

File: Java/Problem_076.java
public class Euler76 {
    public static void main(String[] args) {
        final int N = 100;
        long[] dp = new long[N + 1];
        dp[0] = 1;

        for (int part = 1; part <= 99; part++) {
            for (int s = part; s <= N; s++) {
                dp[s] += dp[s - part];
            }
        }

        System.out.println(dp[N]); // 190569291
    }
}
